src/Features/LanguageServer/Protocol/Handler/InlineCompletions/InlineCompletionsHandler.cs

```diff
@@ -0,0 +1,288 @@
+internal partial class InlineCompletionsHandler : AbstractStatelessRequestHandler<VSInternalInlineCompletionRequest, VSInternalInlineCompletionList>
+{
+    /// <summary>
+    /// The set of built in snippets from, typically found in
+    /// C:\Program Files\Microsoft Visual Studio\2022\VS_INSTANCE\VC#\Snippets\1033\Visual C#
+    /// These are currently the only snippets supported.
+    /// </summary>
+    public static ImmutableHashSet<string> BuiltInSnippets = ImmutableHashSet.Create(
+        "~", "Attribute", "checked", "class", "ctor", "cw", "do", "else", "enum", "equals", "Exception", "for", "foreach", "forr",
+        "if", "indexer", "interface", "invoke", "iterator", "iterindex", "lock", "mbox", "namespace", "#if", "#region", "prop",
+        "propfull", "propg", "sim", "struct", "svm", "switch", "try", "tryf", "unchecked", "unsafe", "using", "while");
+
+    [ImportingConstructor]
+    [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+    public InlineCompletionsHandler()
+    {
+    }
+
+    public override string Method => VSInternalMethods.TextDocumentInlineCompletionName;
+
+    public override bool MutatesSolutionState => false;
+
+    public override bool RequiresLSPSolution => true;
+
+    public override TextDocumentIdentifier? GetTextDocumentIdentifier(VSInternalInlineCompletionRequest request)
+    {
+        return request.TextDocument;
+    }
+
+    public override async Task<VSInternalInlineCompletionList> HandleRequestAsync(VSInternalInlineCompletionRequest request, RequestContext context, CancellationToken cancellationToken)
+    {
+        Contract.ThrowIfNull(context.Document);
+
+        // First get available snippets if any.
+        var snippetInfoService = context.Document.Project.GetRequiredLanguageService<ISnippetInfoService>();
+        var snippetInfo = snippetInfoService.GetSnippetsIfAvailable();
+        if (!snippetInfo.Any())
+        {
+            return new VSInternalInlineCompletionList();
+        }
+
+        // Then attempt to get the word at the requested position.
+        var sourceText = await context.Document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+        var syntaxFactsService = context.Document.Project.GetRequiredLanguageService<ISyntaxFactsService>();
+        var linePosition = ProtocolConversions.PositionToLinePosition(request.Position);
+        var position = sourceText.Lines.GetPosition(linePosition);
+        if (!SnippetUtilities.TryGetWordOnLeft(position, sourceText, syntaxFactsService, out var wordOnLeft))
+        {
+            return new VSInternalInlineCompletionList();
+        }
+
+        // Find the snippet with shortcut text that matches the typed word.
+        var wordText = sourceText.GetSubText(wordOnLeft.Value).ToString();
+        if (!BuiltInSnippets.Contains(wordText, StringComparer.OrdinalIgnoreCase))
+        {
+            return new VSInternalInlineCompletionList();
```

Comment:
Ah does the API not support returning `null`? If it does it might be clear to return that here as a way to very clearly say "we have nothing". Same with above